susemanager-utils/mgr-libmod/mgrlibmod/mllib.py

```diff
@@ -0,0 +1,330 @@
+        module = self._mod_index.get_module(name)
+        if module:
+            for s_obj in module.get_all_streams():
+                streams.add(s_obj.get_stream_name())
+
+        return list(streams)
+
+    def get_stream_contexts(self, s_type: mltypes.MLStreamType) -> List:
+        if self._mod_index is None:
+            self.index_modules()
+        assert self._mod_index is not None, "Unable to get stream contexts: module index not found"
+        contexts: List = []
+        module = self._mod_index.get_module(s_type.name)
+        if module:
+            for stream in module.get_all_streams():
+                if stream.get_stream_name() == s_type.stream:
+                    contexts.append(stream)
+
+        return contexts
+
+    def get_stream_dependencies(self, ctx: Modulemd.ModuleStreamV2) -> List[str]:
+        """
+        get_stream_dependencies -- get stream dependencies.
+
+        :param ctx: module stream context
+        :type ctx: Modulemd.ModuleStreamV2
+        :return: list of dependency names.
+        :rtype: List[str]
+        """
+        deps: List[str] = []
+        s_deps: List[Modulemd.Dependencies] = ctx.get_dependencies() or []
+        dep: Modulemd.Dependencies
+        for dep in s_deps:
+            deps.extend(dep.get_runtime_modules())
+
+        return deps
+
+    def index_modules(self) -> None:
+        """
+        index_modules -- loads given metadata and indexes modules from there.
+        """
+        if self._mod_index is None:
+            mgr: Modulemd.ModuleIndex = Modulemd.ModuleIndexMerger.new()
+            for path in self.metadata:
+                idx = Modulemd.ModuleIndex.new()
+                if self._is_meta_compressed(path):
+                    with gzip.open(path) as gzmeta:
+                        idx.update_from_string(gzmeta.read().decode("utf-8"), True)
+                else:
+                    idx.update_from_file(path, True)
+                mgr.associate_index(idx, 0)
+            self._mod_index = mgr.resolve()
+
+    def get_default_stream(self, name: str):
+        assert self._mod_index is not None, "Unable to access module index when resolving default stream"
+        module = self._mod_index.get_module(name)
+        if not module:
+            raise ValueError("Module {} not found".format(name))
+        defaults = module.get_detaults()
+        if defaults:
+            return defaults.get_default_stream()
+
+        return module.get_all_streams()[0].get_stream_name()
+
+    def get_dep_streams(self, s_obj):
+        dep = s_obj.get_dependencies()[0]  # XXX: Why just always first?
+        all_deps = []  # type: ignore
+        for m in dep.get_runtime_modules():
+            deps = dep.get_runtime_streams(m)
+            if deps:
+                all_deps.append((m, deps[0],))  # XXX: why just first?
+        return all_deps
+
+    def get_actual_stream(self, name: str):
+        if name == "platform":
+            return "e18"
```

Comment:
This should be `el8`, not `e18` (stands for Enterprise Linux 8).

The module named "platform" is a pseudo-module to specify the actual OS in the modules world. So if a module depends on `el:8`, it just means it's built for RHEL 8.

I'm not completely happy hardcoding this here, the values will surely change in the future.

Reply:
🤦‍♂ Right...

Reply:
It is WP. I will surely update this better, once it works at all properly.